Gems/Atom/Feature/Common/Assets/Shaders/Reflections/ReflectionScreenSpaceRayTracingCommon.azsli

```diff
@@ -0,0 +1,698 @@
+    float3 dirToLight = normalize(posToLight);
+    float d2 = dot(posToLight, posToLight); // light distance squared
+    float falloff = d2 * light.m_invAttenuationRadiusSquared;
+    
+    // test for out of range
+    if (falloff >= 1.0f)
+        return;
+
+    float tmax = (sqrt(d2) - lightingData.m_viewBias);
+    if (!CheckLightVisibility(lightingData, dirToLight, tmax))
+    {
+        return;
+    }
+
+    // smoothly adjusts the light intensity so it reaches 0 at light.m_attenuationRadius distance
+    float radiusAttenuation = 1.0f - (falloff * falloff);
+    radiusAttenuation = radiusAttenuation * radiusAttenuation;
+        
+    // standard quadratic falloff
+    d2 = max(0.001f * 0.001f, d2);
+    float3 lightIntensity = (light.m_rgbIntensityCandelas / d2) * radiusAttenuation;
+
+    diffuse += DiffuseLambertian(lightingData.m_albedo, lightingData.m_normal, dirToLight, lightingData.m_diffuseResponse) * lightIntensity;
+
+    if (applySpecular)
+    {
+        specular += SpecularGGX(lightingData.m_dirToCamera, dirToLight, lightingData.m_normal, lightingData.m_specularF0, lightingData.m_NdotV, lightingData.m_roughnessA2, 1.0f) * lightIntensity;
+    }
+}
+
+void ApplySimpleSpotLight(RayTracingSceneSrg::SimpleSpotLight light, RayTracingLightingData lightingData, bool applySpecular, out float3 diffuse, out float3 specular)
+{
+    float3 posToLight = light.m_position - lightingData.m_worldPosition;
+    float3 dirToLight = normalize(posToLight);
+    float d2 = dot(posToLight, posToLight);
+    float falloff = d2 * light.m_invAttenuationRadiusSquared;
+    
+    // test for out of range
+    if (falloff >= 1.0f)
+        return;
+
+    // test for outside of outer cone
+    float dotWithDirection = dot(dirToLight, -normalize(light.m_direction));
+    if (dotWithDirection <= light.m_cosOuterConeAngle)
+        return;
+
+    float tmax = (sqrt(d2) - lightingData.m_viewBias);
+    if (!CheckLightVisibility(lightingData, dirToLight, tmax))
+    {
+        return;
+    }
+
+    // smoothly adjusts the light intensity so it reaches 0 at light.m_attenuationRadius distance
+    float radiusAttenuation = 1.0f - (falloff * falloff);
+    radiusAttenuation = radiusAttenuation * radiusAttenuation;
+
+    // standard quadratic falloff
+    d2 = max(0.001f * 0.001f, d2);
+    float3 lightIntensity = (light.m_rgbIntensityCandelas / d2) * radiusAttenuation;
+
+    diffuse += DiffuseLambertian(lightingData.m_albedo, lightingData.m_normal, dirToLight, lightingData.m_diffuseResponse) * lightIntensity;
+
+    if (applySpecular)
+    {
+        specular += SpecularGGX(lightingData.m_dirToCamera, dirToLight, lightingData.m_normal, lightingData.m_specularF0, lightingData.m_NdotV, lightingData.m_roughnessA2, 1.0f) * lightIntensity;
+    }
+}
+
+void ApplyPointLight(RayTracingSceneSrg::PointLight light, RayTracingLightingData lightingData, bool applySpecular, out float3 diffuse, out float3 specular)
```

Comment:
Are these light calculations different from the ones we run for the forward pass? Just wondering if it can be refactored further into common functions

Reply:
Yes it's the same issue we've run into with other systems such as Hair.  The lighting shader code can't easily be reused right now due to embedded assumptions about Srgs and shader options.  It should be refactored at some point so we can share it.